dev/service-registry/client-core/src/main/scala/com/lightbend/lagom/internal/registry/DevModeSimpleServiceDiscovery.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package com.lightbend.lagom.internal.registry
+
+import java.net.URI
+
+import akka.actor.ActorSystem
+import akka.discovery.SimpleServiceDiscovery._
+import akka.discovery.{ Lookup, ServiceDiscovery, SimpleServiceDiscovery }
+
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.{ ExecutionContext, Future, Promise }
+
+private[lagom] class DevModeSimpleServiceDiscovery(system: ActorSystem) extends SimpleServiceDiscovery {
+  private val clientPromise = Promise[ServiceRegistryClient]
+
+  implicit private val ec: ExecutionContext = system.dispatcher
+
+  def setServiceRegistryClient(client: ServiceRegistryClient): Unit = clientPromise.success(client)
+
+  override def lookup(lookup: Lookup, resolveTimeout: FiniteDuration): Future[Resolved] =
+    for {
+      client <- clientPromise.future
+      uris <- client.locateAll(lookup.serviceName)
+    } yield Resolved(lookup.serviceName, uris.map(toResolvedTarget))
+
+  private def toResolvedTarget(uri: URI) = ResolvedTarget(uri.getHost, optionalPort(uri.getPort))
+
+  private def optionalPort(port: Int): Option[Int] = if (port < 0) None else Some(port)
```

Comment:
Not sure if port 0 is okey...?

Reply:
Won't hurt to expose it if we get it I guess

Reply:
This is handling a quirk of the `java.net.URI` API.

Here's the doc for [`getPort()`](https://docs.oracle.com/javase/8/docs/api/java/net/URI.html#getPort--)

> Returns the port number of this URI.
> The port component of a URI, if defined, is a non-negative integer.
>
> Returns:
> The port component of this URI, or -1 if the port is undefined

So I think in this case, if the port is present, we want to represent it as it appears in the URL (even if it's 0, which is apparently allowed):

```
scala> val u = java.net.URI.create("http://localhost:0")
u: java.net.URI = http://localhost:0

scala> u.getPort
res0: Int = 0
```

If the port is absent, we want to translate it to `None`

```
scala> val u = java.net.URI.create("http://localhost")
u: java.net.URI = http://localhost

scala> u.getPort
res1: Int = -1
```

If it's present but negative… looks like that's allowed, too, but can't be distinguished:

```
scala> val u = java.net.URI.create("http://localhost:-1")
u: java.net.URI = http://localhost:-1

scala> u.getPort
res2: Int = -1
```

I think we'll just have to rely on the assumption that it won't happen in practice because it is invalid. I don't think it's the job of this class to try to validate the data in the registry. It's all internal code anyway, and I'd like to eventually replace the registry with something that takes named (host, port) pairs instead of URIs.

Reply:
Sounds good